File: app/src/main/java/com/hfad/avc/ui/list/PhoneListAdapter.java
package com.hfad.avc.ui.list;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.hfad.avc.R;
import com.hfad.avc.dagger.ComponentManager;

import java.util.ArrayList;

import javax.inject.Inject;

public class PhoneListAdapter extends RecyclerView.Adapter<PhoneListAdapter.ViewHolder> {

   private ArrayList<String> phonelist;
   private Click click;
   @Inject
   Context context;

   public PhoneListAdapter(ArrayList<String> phonelist) {
      ComponentManager.Companion.getInstance().appComponent.inject(this);
      this.phonelist = phonelist;
   }

   @Override
   public PhoneListAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
      View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.phonenomber_item, parent, false);
      return new ViewHolder(view);
   }

   public void setClick(Click click) {
      this.click = click;
   }

   @Override
   public void onBindViewHolder(PhoneListAdapter.ViewHolder holder, int position) {
      holder.phone.setText(phonelist.get(position));
      holder.itemView.setOnClickListener(v -> {

      });
   }

   @Override
   public int getItemCount() {
      return phonelist != null ? phonelist.size() : 0;
   }

   public static class ViewHolder extends RecyclerView.ViewHolder {
      final TextView phone;

      ViewHolder(View view) {
         super(view);
         phone = view.findViewById(R.id.phone);
      }
   }
   public interface Click {
      void click();
   }
}

File: app/src/main/java/com/lastaurus/automatic_congratulations/ui/list/PhoneListAdapter.kt
package com.lastaurus.automatic_congratulations.ui.list

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.lastaurus.automatic_congratulations.R
import com.lastaurus.automatic_congratulations.dagger.ComponentManager.Companion.instance
import javax.inject.Inject

class PhoneListAdapter(private val phonelist: ArrayList<String>) :
   RecyclerView.Adapter<PhoneListAdapter.ViewHolder>() {


   @Inject
   lateinit var context: Context

   init {
      instance.appComponent.inject(this)
   }

   override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
      val view: View =
         LayoutInflater.from(parent.context).inflate(R.layout.phonenomber_item, parent, false)
      return ViewHolder(view)
   }

   override fun onBindViewHolder(holder: ViewHolder, position: Int) {
      holder.phone.text = phonelist[position]
   }

   override fun getItemCount(): Int {
      return phonelist.size
   }

   class ViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(view) {
      val phone: TextView

      init {
         phone = view.findViewById(R.id.phone)
      }
   }
}